vpr/src/base/netlist_walker.cpp

```diff
@@ -10,43 +10,59 @@ void NetlistWalker::walk() {
     visitor_.visit_top(atom_ctx.nlist.netlist_name().c_str());
 
     for(auto blk_id : cluster_ctx.clb_nlist.blocks()) {
-        //Visit the top-level block
-		visitor_.visit_clb(cluster_ctx.clb_nlist.block_pb(blk_id));
+        const auto *pb = cluster_ctx.clb_nlist.block_pb(blk_id);
 
+        //Visit the top-level block
+        visitor_.visit_clb(blk_id, pb);
         //Visit all the block's primitives
-        walk_atoms(cluster_ctx.clb_nlist.block_pb(blk_id));
+        walk_blocks(pb->pb_route, pb, pb->pb_graph_node);
     }
 
     visitor_.finish();
 }
 
-void NetlistWalker::walk_atoms(const t_pb* pb) {
-    //Recursively travers this pb calling visitor_.visit_atom()
-    //on any of its primitive pb's
+void NetlistWalker::walk_blocks(const t_pb_routes &top_pb_route, const t_pb *pb, const t_pb_graph_node *pb_graph_node) {
```

Comment:
Do we need to pass the `pb_graph_node` independently here? I think it should always be accessible as pb->pb_graph_node.

Is there any scenario where the passed pb_graph_node wouldn't match pb->pb_graph_node? That would justify passing it independently. If that is the case there should probably be a comment outlining this scenario, and the invariant are being enforced by walk_blocks to account for it.

Reply:
I believe you are correct.  This code was from when I was trying to get genfasm to work.  The current versions doesn't uses this null traversal, so I went back and removed it.  Thanks!